main/src/ca/uwaterloo/flix/language/phase/Resolver.scala

```diff
@@ -1842,7 +1842,17 @@ object Resolver {
       case NamedAst.Pattern.Tag(qname, pats, loc) =>
         lookupTag(qname, env, ns0, root) match {
           case Result.Ok(c) =>
-            val ps = pats.map(visit)
+            val ps0 = pats.map(visit)
+            // If there is not a pattern for each term, give an error.
+            val ps = if (ps0.lengthCompare(c.tpes) != 0) {
+              val expectedTerms = c.tpes.length
+              val error = ResolutionError.MismatchedTagPatternArity(c.sym, expected = expectedTerms, actual = ps0.length, loc)
+              sctx.errors.add(error)
+              // maintain as many sensible arguments to allow further type checking, etc.
+              ps0.take(expectedTerms).padTo(expectedTerms, ResolvedAst.Pattern.Wild(loc.asSynthetic))
```

Comment:
If this padding/trimming is not done, then the user will see both the new pretty error and the old type error. With this added you might now just get undefined variables (gave too many, so the last extra are ignored)

Reply:
Why Pattern.Wild and not Pattern.Error?

Reply:
fixed